### PR TITLE
Mobile view adjustments & more

### PR DESCRIPTION
**Pull request by FilipLeitner**
### Mobile view adjustments such as:
- enlarged root font-size (in combination with rem units on components) to adjust the UI. This however not always desired (when HSL is used only as a part of the page and mobile view is triggered). Font size change is configurable by css variable `--hs-mobile-view-font-size-base`. Default is 140%.
-  sidebar is no longer resized by clicking on expander but by dragging the UI component that replaced the expander. Dragged sidebar then toggles between two predefined states (default and collapsed) which are configurable by css variables 
```
  --hs-panelspace-normal-height: //defaults to 70vh
  --hs-panelspace-collapsed-height: //defaults to 20vh
```

https://github.com/user-attachments/assets/5515a230-2be5-4f46-8089-9a34314bcd9a
- Max number of sidebar-items in mobile view sidebar is now 6 to keep it resonably accessible
- some more

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

**Comment by FilipLeitner** [comment]
Ill try to fix the failing test but its a bit annoying that the error only happen on github. Localy on node v22 & v20 everything runs well

**Comment by FilipLeitner** [comment]
> Ill try to fix the failing test but its a bit annoying that the error only happen on github. Localy on node v22 & v20 everything runs well

Really annoying. https://github.com/angular/angular/issues/54527

File: projects/hslayers/core/hslayers.component.ts [hunk 1]
```diff
@@ -40,6 +49,20 @@ export class HslayersComponent implements AfterViewInit, OnInit {
...
 
+  private panState: PanState = {
+    /**
+     * Maximum and minimum values of how much the panel can 'moved' by panning
+     * Final values are controlled by css variables
+     */
+    MIN_HEIGHT: 20, // vh
+    MAX_HEIGHT: 70, // vh
+    isProcessing: false,
+    TOGGLE_THRESHOLD: 200, // px
```
Comment (jmacura) [inline projects/hslayers/core/hslayers.component.ts:60]:
```suggestion
    /**
     * Threshold in px
     */
    TOGGLE_THRESHOLD: 200,
```

File: projects/hslayers/core/hslayers.component.ts [hunk 2]
```diff
@@ -40,6 +49,20 @@ export class HslayersComponent implements AfterViewInit, OnInit {
   sidebarVisible: boolean;
   private destroyRef = inject(DestroyRef);
 
+  private panState: PanState = {
+    /**
+     * Maximum and minimum values of how much the panel can 'moved' by panning
+     * Final values are controlled by css variables
+     */
```
Comment (jmacura) [inline projects/hslayers/core/hslayers.component.ts:56]:
```suggestion
    /**
     * Maximum and minimum values in 'vh' units of how much the panel can be 'moved' by panning
     * Final values are controlled by css variables
     */
```

**Review by jmacura — COMMENTED** [review]
A lot of great and tricky changes, thanks!

Please check my comments regarding the logical OR vs. nullish coalescing.

File: projects/hslayers/services/add-data/url/arcgis.service.ts [hunk 3]
```diff
@@ -304,6 +304,7 @@ export class HsUrlArcGisService implements HsUrlTypeServiceModel {
         : await this.calcAllLayersExtent(layers, options);
 
     const layerParams = {
+      opacity: options.opacity || 1,
```
[the remark this answers is not in the document]
Comment (FilipLeitner) [reply projects/hslayers/services/add-data/url/arcgis.service.ts]: [3ab8607](https://github.com/hslayers/hslayers-ng/pull/5432/commits/3ab8607545bfc3628dc0794beb57123e812dc6b7)

**Pull request merged**